includes/class-content-toc.php

```diff
@@ -134,10 +135,12 @@ public function shortcode( $atts, $content = null ) {
 
 			$toc_html .= apply_filters(
 				'hm_content_toc_list',
-				'<ul id="hm-content-toc-' . esc_attr( $this->id_counter ) . '" class="hm-content-toc">' . $toc_list . '</ul>',
+				'<ul class="hm-content-toc">' . $toc_list . '</ul>',
```

Comment:
Easiest way is probably to stitch it together using `sprintf()` statements that take variables from a settings array like in widgets eg.

```php
$settings = array(
     'wrapper_class' => 'hm-toc-wrapper',
     'list_tag' => 'ul',
     'list_item_tag' => 'li',
     'list_class' => 'hm-toc-list',
     'title' => 'Table of contents',
     'title_tag' => 'h3',
     'title_class' => 'hm-toc-title'
);

$settings = apply_filters( 'hm_toc_settings', $settings, $post_id );
```

So you could then use those vars throughout the HTML construction in `sprintf()` calls or just in string concatenation eg:

```php
if ( $settings['title'] ) {
  $toc_title = sprintf( '<%1$s class="%2$s">%3$s</%1$s>', $settings['title_tag'], esc_attr( $settings['title_class'] ), esc_html( $settings['title'] ) );
  $output .= $toc_title;
}
```

Essentially you have one place for settings / config that you filter and then use the bits of it in the rest of the function rather than at the point of use. Does that help?